fence/agents/ironic/fence_ironic.py

```diff
@@ -0,0 +1,138 @@
+#!@PYTHON@ -tt
+
+import atexit
+import logging
+import os
+import re
+import sys
+from pipes import quote
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import fail_usage, is_executable, run_command, run_delay
+
+#BEGIN_VERSION_GENERATION
+RELEASE_VERSION=""
+REDHAT_COPYRIGHT=""
+BUILD_DATE=""
+#END_VERSION_GENERATION
+
+def get_name_or_uuid(options):
+	return options["--uuid"] if "--uuid" in options else options["--plug"]
+
+def get_power_status(_, options):
+    output = ironic_run_command(options, "status")
+    stdout = output[1]
+    match = re.search('power[\\s]*([a-zA-Z]{2,3})', str(stdout))
+    status = match.group(1) if match else None
+    return status
+
+def set_power_status(_, options):
+    ironic_run_command(options, options["--action"])
+    return
+
+def get_devices_list(_, options):
+    nodes = {}
+    output = ironic_run_command(options, "list")
+    stdout = output[1]
+    for line in stdout.splitlines():
+        try:
+            (uuid, name, state) = line.split(',')
+        except ValueError:
+            pass
+        if "UUID" in uuid:
+            continue # skip line header
+        match = re.search('power[\\s]*([a-zA-Z]{2,3})', state)
+        status = match.group(1) if match else None
+        nodes[uuid] = (name, status)
+
+    return nodes
+
+def ironic_run_command(options, action, timeout=None):
+    cmd = options["--openstackpath"] + " baremetal node"
+    env = os.environ.copy()
+    # --username / -l
+    if "--username" in options and len(options["--username"]) != 0:
+        env["OS_USERNAME"] = options["--username"]
+
+    # --password / -p
+    if "--password" in options:
+        x = quote(options["--password"])
+        x = x[:-1] if x.endswith("'") else x
+        x = x[1:] if x.startswith("'") else x
+        env["OS_PASSWORD"] = x
+
+    # --tenant-name -t
+    if "--tenant-name" in options:
+        env["OS_TENANT_NAME"] = options["--tenant-name"]
+
+    #  --auth-url -k
+    if "--auth-url" in options:
+        env["OS_AUTH_URL"] = options["--auth-url"]
+
+    # --action / -o
+    if action == "status":
+        cmd += " show %s -c power_state --format value" % (get_name_or_uuid(options))
+    elif action in ["on", "off"]:
+        cmd += " power %s %s" % (action, get_name_or_uuid(options))
+    elif action == "list":
+        cmd += " list -c UUID -c Name -c 'Power State' --format csv --quote minimal"
+
+
+    logging.debug("cmd -> %s" % cmd)
+    return run_command(options, cmd, timeout, env)
+
+def define_new_opts():
+    all_opt["auth-url"] = {
+        "getopt" : "k:",
```

Comment:
Please, do not introduce short options as we already have too many of them. That's the problem on our side.